### PR TITLE
sdi-461 identify admissions due to an offender merge

### PR DESCRIPTION
**Pull request by lbennett-moj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/diff/PrisonerMovementsEventService.kt [hunk 1]
```diff
@@ -180,7 +208,9 @@ sealed class PossibleMovementChange {
     class TransferOut(offenderNo: String, prisonId: String) :
       MovementOutChange(offenderNo, prisonId, PrisonerReleaseReason.TRANSFERRED)
 
-    class TAPRelease(offenderNo: String, prisonId: String) : MovementOutChange(offenderNo, prisonId, TEMPORARY_ABSENCE_RELEASE)
+    class TAPRelease(offenderNo: String, prisonId: String) :
+      MovementOutChange(offenderNo, prisonId, TEMPORARY_ABSENCE_RELEASE)
```
Comment (lbennett-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/diff/PrisonerMovementsEventService.kt:212]: formatting

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/diff/PrisonerMovementsEventService.kt [hunk 2]
```diff
@@ -159,6 +171,13 @@ private fun Prisoner.isSomeOtherMovementOut(previousPrisonerSnapshot: Prisoner?)
   this.inOutStatus == "OUT" &&
     this.status != previousPrisonerSnapshot?.status
 
+private fun isAdmissionAssociatedWithAMerge(offenderBooking: OffenderBooking): Boolean {
+  return offenderBooking.identifiers?.filter { it.type == "MERGED" }?.sortedByDescending { it.whenCreated }?.first()
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/diff/PrisonerMovementsEventService.kt:175]: very minor but you could have done 
`offenderBooking.identifiers?.filter { it.type == "MERGED" }?.any(it.whenCreated > LocalDateTime.now().minusMinutes(90))`

**Review by andymarke — DISMISSED** [review]
LGTM

**Review by andymarke — APPROVED** [review]
lgtm

**Pull request merged**